strawberry/federation/field.py

```diff
@@ -87,7 +86,7 @@ def field(
     deprecation_reason: Optional[str] = None,
     default: Any = UNSET,
     default_factory: Union[Callable, object] = UNSET,
-    directives: Sequence[StrawberrySchemaDirective] = (),
```

Comment:
I'm pretty confused by the loosening of the types everywhere

Reply:
That's because the current schema directives are not `StrawberrySchemaDirectives` anymore, they are dataclasses that don't derive from anything.

Consider this example:

```python
import strawberry
from strawberry.schema_directive import Location


@strawberry.schema_directive(locations=[Location.OBJECT])
class SomeSchemaDirective:
    foo: str
    bar: int


@strawberry.type(
    directives=[SomeSchemaDirective(foo="foo", bar=0)],
)
class SomeType:
    bin: str
```

Without this change, it is impossible to use this code without having to `type:ignore` pyright/mypy/etc. For example, running pyright on that code:

```
/tmp/xxx/xxx.py
  /tmp/xxx/xxx.py:12:17 - error: Expected no arguments to "SomeSchemaDirective" constructor (reportGeneralTypeIssues)
  /tmp/xxx/xxx.py:12:16 - error: Argument of type "list[SomeSchemaDirective]" cannot be assigned to parameter "directives" of type "Sequence[StrawberrySchemaDirective] | None" in function "type"
    Type "list[SomeSchemaDirective]" cannot be assigned to type "Sequence[StrawberrySchemaDirective] | None"
      TypeVar "_T_co@Sequence" is covariant
        "SomeSchemaDirective" is incompatible with "StrawberrySchemaDirective"
      Type cannot be assigned to type "None" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations 
```

Reply:
that's correct, unfortunately we can't do much to improve the typing for directives (and types) for the time being